docs/proposals/nginx-extensions.md

```diff
@@ -0,0 +1,964 @@
+| Retries. This is different from NGINX retries | Open         | [GEP-1731](https://github.com/kubernetes-sigs/gateway-api/issues/1731)   |
+| Session persistence                           | Provisional  | [GEP-1619](https://gateway-api.sigs.k8s.io/geps/gep-1619/)               |
+| Rate-limiting                                 | Open         | [Issue 326](https://github.com/kubernetes-sigs/gateway-api/issues/326)   |
+| Cross-Origin Resource Sharing (CORS)          | Open         | [Issue 1767](https://github.com/kubernetes-sigs/gateway-api/issues/1767) |
+| Gateway client certificate verification       | Provisional  | [GEP-91](https://gateway-api.sigs.k8s.io/geps/gep-91/)                   |
+| Authentication                                | Open         | [Issue 1494](https://github.com/kubernetes-sigs/gateway-api/issues/1494) |
+
+## Grouping the Features
+
+To reduce the number of CRDs NGINX Gateway Fabric must maintain and users have to create, we grouped the high and medium-priority features into configuration categories.
+
+For each group, we will begin implementation by focusing on the high-priority features of each group. Then, once the high-priority features are complete, we can move on to the medium-priority features.
+
+The low-priority features are out of scope for the rest of the Enhancement Proposal but may be revisited once we make progress on the higher-priority features. In addition, most features with active GEPs are not included in the groups as we want to help move the GEPs forward toward standardization instead of creating our bespoke solutions.
+
+When grouping the features, we considered the following factors:
+
+1. Use cases. Are there features that can be grouped by the use case they satisfy? For example, authentication or observability.
+2. Gateway API roles. Which features are the domain of the Cluster Operator? Which features are Application Developers responsible for?
+3. The NGINX contexts. Which contexts are the NGINX directives available in? For example, http, server, location, upstream, etc.
+
+The following picture shows the nine groups we came up with:
+
+![groups](/docs/images/nginx-functionality-groups.png)
+
+## API
+
+The following section proposes an extension type (Policy, Filter, etc.), extension point (GatewayClass, HTTPRoute, etc.), and Gateway API role for all the groups. The goal of this Enhancement Proposal is not to design each extension but to propose a strategy for delivering critical NGINX features to our customers.
+
+### Gateway Settings
+
+_Extension type:_ ParametersRef
+
+_Resource type:_ CRD
+
+_Role(s):_ Cluster Operator
+
+_Extension point:_ GatewayClass or Gateway
+
+_NGINX Context(s):_ main, http, stream
+
+Features:
+
+- Error log level
+- Access log settings: format and disable
+- PROXY protocol
+- OTEL Tracing: global settings such as exporter configuration and global enable/disable.
```

Comment:
Does global enable/disable creates a sort-of like inherited policy, considering also the proposed Observability Policy ?

Reply:
Yes, it kind of does, but I think it makes sense for the Cluster Operator to be able to allow tracing, and then App Devs can choose to disable it for their applications. 

Alternatively, we could omit the enable field here, and if an exporter is provided in the Gateway Settings, we effectively enable tracing it for all the Routes.

Then, in the Observability Policy, we add a `disable` field that allows App Devs to opt out of tracing but doesn't allow them to enable it. 

Would that be better?

Reply:
> Alternatively, we could omit the enable field here, and if an exporter is provided in the Gateway Settings, we effectively enable tracing it for all the Routes.

tracing adds overhead for NGINX. Because of that, I don't think it should be enabled by default.

What if we start with HTTPRoute Observability policies.. and if there is demand to support enabling tracing by globally or per listener, we will add that support?

Reply:
That sounds fine. Since the Cluster Operator provides the exporter endpoint, they can still effectively disable tracing for all routes by not providing an endpoint. However, by removing the enable field, they can't enable it globally. Does that sound right to you?

Reply:
> That sounds fine. Since the Cluster Operator provides the exporter endpoint, they can still effectively disable tracing for all routes by not providing an endpoint. However, by removing the enable field, they can't enable it globally. Does that sound right to you?

Let's try to define cases here:

- if tracing is required for all apps, cluster operator can enforce it
- If tracing is not required for all apps,  app devs can selectively enable it
- There is also a case when tracing is enabling by default, but app dev disables it for certain HTTPRoutes.

Can we support those?

Unfortunately, we also don't have user data/requests to understand which case would be most common